src/app/register/register.component.html

```diff
@@ -1,38 +1,79 @@
+
+  <form [formGroup]="regForm" (ngSubmit)="onSubmit(regForm.value)" id="registration">
+    <div class="form-container">
+      <mat-form-field appearance="outline">
+        <mat-label>First Name</mat-label>
+        <mat-icon matSuffix>person</mat-icon>
+        <input matInput name="fname" placeholder={{language?.placeHolders.firstName}}
+               [formControl]="regForm.controls['fname']">
+        <mat-error *ngIf="regForm.controls['fname'].hasError('required')">{{authMessages?.errors.firstNameReq}}
+        </mat-error>
+      </mat-form-field>
+      <mat-form-field appearance="outline">
+        <mat-label>Second Name</mat-label>
+        <mat-icon matSuffix>person</mat-icon>
+        <input matInput name="lname" placeholder={{language?.placeHolders.lastName}}
+               [formControl]="regForm.controls['lname']">
+      </mat-form-field>
+      <mat-form-field appearance="outline">
+        <mat-label>E-Mail</mat-label>
+        <mat-icon matSuffix>mail_outline</mat-icon>
+        <input matInput name="email" placeholder={{language?.placeHolders.email}}
+               [formControl]="regForm.controls['email']">
+        <mat-error *ngIf="regForm.controls['email'].hasError('required')">{{authMessages?.errors.emailReq}}</mat-error>
+        <mat-error *ngIf="regForm.controls['email'].hasError('pattern')">{{authMessages?.errors.emailInvalid}}
+        </mat-error>
+      </mat-form-field>
+      <mat-form-field appearance="outline">
+        <mat-label>Password</mat-label>
+        <mat-icon matSuffix (click)="hide = !hide">{{hide ? 'visibility' : 'visibility_off'}}</mat-icon>
+        <input matInput type="password" [type]="hide ? 'password' : 'text'" name="password"
+               placeholder={{language?.placeHolders.password}} [formControl]="regForm.controls['password']">
+        <mat-error *ngIf="regForm.controls['password'].hasError('required')">{{authMessages?.errors.passwordReq}}
+        </mat-error>
+        <mat-error *ngIf="regForm.controls['password'].hasError('pattern')">{{authMessages?.errors.passwordPattern}}
+        </mat-error>
+        <mat-error *ngIf="regForm.controls['password'].hasError('minlength')">{{authMessages?.errors.passwordLength}}
+        </mat-error>
+      </mat-form-field>
+      <br
+        *ngIf="regForm.controls['password'].hasError('pattern') && regForm.controls['password'].hasError('minlength') ">
+      <mat-form-field appearance="outline">
+        <mat-label>Confirm Password</mat-label>
+        <mat-icon matSuffix (click)="hide = !hide">{{hide ? 'visibility' : 'visibility_off'}}</mat-icon>
+        <input matInput type="password" [type]="hide ? 'password' : 'text'" name="confirmpassword"
+               placeholder={{language?.placeHolders.confirmPassword}}
+               [formControl]="regForm.controls['confirmpassword']">
+        <mat-error *ngIf="regForm.controls['confirmpassword'].value!=regForm.controls['password'].value">
```

Comment:
are these error messages only triggered on submit ?

Reply:
These messages are triggered when you focus out of the input field without typing the constrained test.